### PR TITLE
Update 2024 Mentorship Program Timeline

### PR DESCRIPTION
**Pull request by devanshimodha**
This change is to update the 2024 mentorship timeline.

Please Note: Survey link is yet to be updated

### CODE REVIEW

**Review by 0xTim — APPROVED** [review]

**Comment by 0xTim** [comment]
@swift-ci test

**Review by federicobucchi — APPROVED** [review]

**Comment by alexandersandberg** [comment]
@devanshimodha do we want to wait with merging until the survey link is updated or not? If so, we might want to mark this as a draft in the meantime.

**Comment by devanshimodha** [comment]
@alexandersandberg yes, I would like to add the updated link in the same PR

**Comment by alexandersandberg** [comment]
> @alexandersandberg yes, I would like to add the updated link in the same PR

Ok, marked this as a draft in the meantime so we don't accidentally merge it. 🙂

**Comment by devanshimodha** [comment]
Timeline has been updated along with the survey link which opens at 8:00 AM PT

File: mentorship/index.md [hunk 1]
```diff
@@ -15,19 +15,19 @@ If the mentee has not contributed to the project before, they will first work wi
...
-| **October 27th**   |  Feedback deadline               |
+| **June 17th**      |  Mentor Interest surveys open    |
+| **July 01st**      |  Mentor Interest survey deadline |
+| **July 08th**      |  Mentee Interest surveys open    |
+| **July 15th**      |  Mentee Interest survey deadline |
+| **July 29th**      |  Mentorship pairings announced   |
+| **August 05th**    |  10-week cohort begins           |
+| **September 09th** |  Halfway checkpoint              |
+| **October 14th**   |  10-week cohort ends             |
+| **October 21st**   |  Feedback deadline               |
```
Comment (alexandersandberg) [inline mentorship/index.md:26]: We might not need the number padding here?

```suggestion
| **June 17th**      |  Mentor Interest surveys open    |
| **July 1st**      |  Mentor Interest survey deadline |
| **July 8th**      |  Mentee Interest surveys open    |
| **July 15th**      |  Mentee Interest survey deadline |
| **July 29th**      |  Mentorship pairings announced   |
| **August 5th**    |  10-week cohort begins           |
| **September 9th** |  Halfway checkpoint              |
| **October 14th**   |  10-week cohort ends             |
| **October 21st**   |  Feedback deadline               |
```

File: mentorship/index.md [hunk 2]
```diff
@@ -15,19 +15,19 @@ If the mentee has not contributed to the project before, they will first work wi
...
 
 ### Interest Surveys
 
-The 2023 mentor interest survey submissions are open from June 12th - June 26th. If you are interested in participating as a mentor, please fill out the interest survey [here](https://essentials.applesurveys.com/jfe/form/SV_9vEcwfZtSblGYjs).
+The 2024 mentor interest survey submissions are open from June 17th - July 01st. If you are interested in participating as a mentor, please fill out the interest survey [here](https://essentials.applesurveys.com/jfe/form/SV_3CSIxEKQmL1MVhQ).
```
Comment (alexandersandberg) [inline mentorship/index.md:30]: Same here.

```suggestion
The 2024 mentor interest survey submissions are open from June 17th - July 1st. If you are interested in participating as a mentor, please fill out the interest survey [here](https://essentials.applesurveys.com/jfe/form/SV_3CSIxEKQmL1MVhQ).
```

**Review by alexandersandberg — COMMENTED** [review]
https://github.com/apple/swift-org-website/pull/695/files#diff-ec0696ff3a544f293063a8926c22bcd168f7c02c11fee68498f8328d38d71cb2R14

We need to change this to "2024 Timeline"

https://github.com/apple/swift-org-website/pull/695/files#diff-ec0696ff3a544f293063a8926c22bcd168f7c02c11fee68498f8328d38d71cb2R80

We should change this to 10 weeks.

**Review by alexandersandberg — APPROVED** [review]
👍

**Comment by alexandersandberg** [comment]
@swift-ci please test

**Comment by alexandersandberg** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@alexandersandberg Please merge this when you have a change, I don't have sufficient permissions to do so. Also pinged @shahmishal about this.

**Comment by alexandersandberg** [comment]
@xedin Done! ✅

**Comment by xedin** [comment]
Thank you!

**Pull request merged**